include/aos/iam/provisionmanager.hpp

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_PROVISIONMANAGER_HPP_
+#define AOS_PROVISIONMANAGER_HPP_
+
+#include <aos/common/tools/error.hpp>
+
+#include "aos/iam/certhandler.hpp"
+
+namespace aos::iam::provisionmanager {
+
+/** @addtogroup iam Identification and Access Manager
+ *  @{
+ */
+
+/**
+ * Certificate type configuration.
+ */
+struct CertTypeConfig {
```

Comment:
* Extend existing `aos::iam::certhandler::ModuleConfig` with `mIsSelfSigned` field;
* Extend and implement `CertHandlerItf` to retrieve the module config or to check if specific storage type is self-signed;
* Add parsing this field in aos_core_iam_cpp.